src/render.rs

```diff
@@ -64,68 +64,79 @@ impl fmt::Display for HalfMove {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
         match self {
             HalfMove::HalfMove(d) => {
-                let departure = match d.departure {
-                    None => "".to_string(),
-                    Some(ref dep) => format!("{}", dep),
+                write!(f, "{}", d.figure)?;
+                match d.departure {
+                    None => (),
+                    Some(ref dep) => {
+                        write!(f, "{}", dep)?;
+                    }
                 };
-                write!(
-                    f,
-                    "{}{}{}{}{}",
-                    d.figure,
-                    departure,
-                    if d.capture { "x" } else { "" },
-                    d.arrival,
-                    if d.check { "+" } else { "" }
-                )
-                .unwrap();
+                if d.capture {
+                    write!(f, "x")?;
+                }
+                write!(f, "{}", d.arrival)?;
+                if d.check {
+                    write!(f, "+")?;
+                }
                 if d.en_passant {
-                    write!(f, " e.p.").unwrap()
-                };
+                    write!(f, " e.p.")?;
+                }
                 if d.draw_offer {
-                    write!(f, " (=)")
-                } else {
-                    Ok(())
+                    write!(f, " (=)")?;
                 }
+
+                Ok(())
+            }
+            HalfMove::KingsideCastling => {
+                write!(f, "0-0")
+            }
+            HalfMove::QueensideCastling => {
+                write!(f, "0-0-0")
             }
-            HalfMove::KingsideCastling => write!(f, "0-0"),
-            HalfMove::QueensideCastling => write!(f, "0-0-0"),
         }
     }
 }
 
-fn render_move(n: usize, m: &Move) -> String {
-    format!("{}. {} {}", n, m.white, m.black)
+impl fmt::Display for Move {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{} {}", self.white, self.black)
+    }
 }
 
-fn score_by_color(color: Color) -> String {
-    match color {
-        Color::White => "1-0".to_string(),
-        Color::Black => "0-1".to_string(),
+impl fmt::Display for Color {
```

Comment:
@snoyberg this doesn't look right but I will rewrite the code to use `impl fmt:Display for Result`

Reply:
I see what you mean, I agree. Good catch.